javav2/example_code/ecr/src/main/java/com/example/ecr/scenario/ECRScenario.java

```diff
@@ -0,0 +1,335 @@
+                   accountId - Your AWS account number. 
+                """;
+
+        if (args.length != 2) {
+            System.out.println(usage);
+            return;
+        }
+
+        ECRActions ecrActions = new ECRActions();
+        String iamRole = args[0];
+        String accountId = args[1];
+        String localImageName;
+
+        Scanner scanner = new Scanner(System.in);
+        System.out.println("""
+            The Amazon Elastic Container Registry (ECR) is a fully-managed Docker container registry 
+            service provided by AWS. It allows developers and organizations to securely 
+            store, manage, and deploy Docker container images. 
+            ECR provides a simple and scalable way to manage container images throughout their lifecycle, 
+            from building and testing to production deployment.\s
+                        
+            The `EcrAsyncClient` interface in the AWS SDK for Java 2.x provides a set of methods to 
+            programmatically interact with the Amazon ECR service. This allows developers to 
+            automate the storage, retrieval, and management of container images as part of their application 
+            deployment pipelines. With ECR, teams can focus on building and deploying their 
+            applications without having to worry about the underlying infrastructure required to 
+            host and manage a container registry.
+            
+           This scenario walks you through how to perform key operations for this service.  
+           Let's get started...
+          """);
+
+        waitForInputToContinue(scanner);
+        System.out.println(DASHES);
+
+        System.out.println("""
+           1. Create an ECR repository.
+            
+           The first task is to ensure we have a local Docker image named echo-text. 
+           If this image exists, then an Amazon ECR repository is created. 
+           
+           An ECR repository is a private Docker container repository provided 
+           by Amazon Web Services (AWS). It is a managed service that makes it easy 
+           to store, manage, and deploy Docker container images.\s
+           """ );
+
+        // Ensure that a local docker image named echo-text exists.
+        boolean doesExist = ecrActions.isEchoTextImagePresent();
+        String repoName;
+        if (!doesExist){
+            System.out.println("The local image named echo-text does not exist");
+            return;
+        } else {
+            localImageName = "echo-text";
+            repoName = "echo-text";
+        }
+
+        try {
+            String repoArn = ecrActions.createECRRepository(repoName);
+            System.out.println("The ARN of the ECR repository is " + repoArn);
+
+        } catch (IllegalArgumentException e) {
+            System.err.println("Invalid repository name: " + e.getMessage());
+            e.printStackTrace();
```

Comment:
The stack traces on these exits probably aren't needed. Is it giving any value to the user?

Reply:
Removed those.